packages/api/lambdas/reports/orca-backup-reconciliation-report.js

```diff
@@ -235,40 +383,54 @@ async function orcaReconciliationReportForGranules(recReportParams) {
   };
 
   const collectionsConfig = await fetchCollectionsConfig(recReportParams);
-  log.debug(`fetchESCollections returned ${JSON.stringify(collectionsConfig)}`);
-
-  const esSearchParams = convertToESGranuleSearchParamsWithCreatedAtRange(recReportParams);
-  log.debug(`Create ES granule iterator with ${JSON.stringify(esSearchParams)}`);
-  const esGranulesIterator = new ESSearchQueue(
-    {
-      ...esSearchParams,
-      sort_key: ['granuleId', 'collectionId'],
-    },
-    'granule',
-    process.env.ES_INDEX
+  log.debug(`fetchCollections returned ${JSON.stringify(collectionsConfig)}`);
+
+  const knex = await getKnexClient();
+  const searchParams = convertToDBGranuleSearchParams(recReportParams);
+
+  const granulesSearchQuery = getGranulesByApiPropertiesQuery({
+    knex,
+    searchParams,
+    sortByFields: ['granule_id', 'collectionName', 'collectionVersion'],
+    temporalBoundByCreatedAt: true,
+  });
+
+  log.debug(`Create PG granule iterator with ${granulesSearchQuery}`);
+
+  const pgGranulesIterator = new QuerySearchClient(
+    granulesSearchQuery,
+    100 // arbitrary limit on how items are fetched at once
   );
 
   const orcaSearchParams = convertToOrcaGranuleSearchParams(recReportParams);
   log.debug(`Create ORCA granule iterator with ${JSON.stringify(orcaSearchParams)}`);
   const orcaGranulesIterator = new ORCASearchCatalogQueue(orcaSearchParams);
 
   try {
+    /** @type {[CumulusGranule, any]} */
+    // @ts-ignore TODO: Ticket CUMULUS-3887 filed to resolve
     let [nextCumulusItem, nextOrcaItem] = await Promise.all(
-      [esGranulesIterator.peek(), orcaGranulesIterator.peek()]
+      [
+        /** @type CumulusGranule */
+        pgGranulesIterator.peek(),
+        orcaGranulesIterator.peek(),
+      ]
     );
 
     while (nextCumulusItem && nextOrcaItem) {
-      const nextCumulusId = `${nextCumulusItem.granuleId}:${nextCumulusItem.collectionId}`;
+      const nextCumulusId = `${nextCumulusItem.granule_id}:${constructCollectionId(nextCumulusItem.collectionName, nextCumulusItem.collectionVersion)}`;
       const nextOrcaId = `${nextOrcaItem.id}:${nextOrcaItem.collectionId}`;
       if (nextCumulusId < nextOrcaId) {
         // Found an item that is only in Cumulus and not in ORCA.
-        addGranuleToReport({
+        // eslint-disable-next-line no-await-in-loop
+        await addGranuleToReport({
```

Comment:
so addGranuleToReport is likely called for every granule to retrieve files?

Reply:
@jennyhliu correct - this will increase the number of database calls, but by making them per-granule given how the rec reports are currently setup this should effectively disperse that load. 

We absolutely need to improve the overall setup for these tests, that work should neatly fall under the 'improve rec reports' epic.